pkg/synthetictests/allowedalerts/basic_alert.go

```diff
@@ -221,36 +221,45 @@ func (a *basicAlertTest) failOrFlake(ctx context.Context, restConfig *rest.Confi
 	return pass, ""
 }
 
+var unrecognizedSignatureRegEx = regexp.MustCompile("reason/ErrImagePull UnrecognizedSignatureFormat")
+
+func kubePodNotReadyDueToErrParsingSignature(trackedEventResources monitorapi.InstanceMap, firingIntervals monitorapi.Intervals) bool {
+	return kubePodNotReadyDueToRegExMatch(trackedEventResources, firingIntervals, unrecognizedSignatureRegEx)
+}
+
 var imagePullBackoffRegEx = regexp.MustCompile("Back-off pulling image .*registry.redhat.io")
 
 // kubePodNotReadyDueToImagePullBackoff returns true if we searched pod events and determined that the
 // KubePodNotReady alert for this pod fired due to an imagePullBackoff event on registry.redhat.io.
 func kubePodNotReadyDueToImagePullBackoff(trackedEventResources monitorapi.InstanceMap, firingIntervals monitorapi.Intervals) bool {
+	return kubePodNotReadyDueToRegExMatch(trackedEventResources, firingIntervals, imagePullBackoffRegEx)
+}
 
+func kubePodNotReadyDueToRegExMatch(trackedEventResources monitorapi.InstanceMap, firingIntervals monitorapi.Intervals, regexp *regexp.Regexp) bool {
 	// Run the check for all firing intervals.
 	for _, firingInterval := range firingIntervals {
 		relatedPodRef := monitorapi.PodFrom(firingInterval.Locator)
 
 		// Find an event
-		foundImagePullBackoffEvent := false
+		foundRegexMatchEvent := false
 		var tmpEvent *corev1.Event
 		for _, obj := range trackedEventResources {
 			tmpEvent = obj.(*corev1.Event)
 			if tmpEvent.InvolvedObject.Name == relatedPodRef.Name &&
 				tmpEvent.InvolvedObject.Namespace == relatedPodRef.Namespace &&
-				imagePullBackoffRegEx.MatchString(tmpEvent.Message) {
-				foundImagePullBackoffEvent = true
+				regexp.MatchString(tmpEvent.Message) {
+				foundRegexMatchEvent = true
 				break
 			}
 		}
-		if !foundImagePullBackoffEvent {
+		if !foundRegexMatchEvent {
 			// No event resources found so we can't do any checking.
 			return false
 		}
-		imagePullBackoffTime := tmpEvent.LastTimestamp.Time
+		regexMatchEventTime := tmpEvent.LastTimestamp.Time
 		alertTime := firingInterval.From
-		if alertTime.After(imagePullBackoffTime) && alertTime.Sub(imagePullBackoffTime) < time.Minute*10 {
-			framework.Logf("KubePodNotReady alert failure suppressed due to registry.redhat.io ImagePullBackoff on pod %s/%s",
+		if alertTime.After(regexMatchEventTime) && alertTime.Sub(regexMatchEventTime) < time.Minute*10 {
+			framework.Logf("KubePodNotReady alert failure suppressed due to %s on pod %s/%s", tmpEvent.Message,
```

Comment:
Can you elaborate on suppression vs a clear test failure regarding the problem? I know both options were discussed at one point but I can't recall why this one won out. I do recall we said we don't care about loki, but this looks like full suppression.

It might be good to have a test failing if we see this problem at all, so we can convey that info to the team doing those fixes for us. (or perhaps flaking)

Reply:
I'm good either way.  I thought suppression was wanted so we didn't have all the failed payloads.  But that seems to be less of an issue now so I can refactor if we just want a specific test to highlight the issue.